### PR TITLE
Cross platform display

### LINKED ISSUE #254: [13] Cross platform trust display with fallback
The trust display needs to be available on all platforms. If the required dependencies are not (yet) available on a platform, a fallback is needed to the old trust screen.

- [x] Windows 32-bit
- [x] Windows 64-bit
- [x] MacOS
- [x] Linux

### PR DESCRIPTION
**Pull request by MaxVanDeursen**
In this PR, a fallback is added if PyQt5.WebEngineWidgets.QWebEngineView cannot be imported to the old trust page. This trust page links to a webpage as well, containing the new GUI.

Fixes #254

### CODE REVIEW

File: TriblerGUI/qt_resources/mainwindow.ui [hunk 1]
```diff
@@ -7735,8 +7735,8 @@ QTabBar::tab:selected {
                       <rect>
                        <x>0</x>
                        <y>0</y>
-                       <width>133</width>
```
Comment (vandenheuvel) [inline TriblerGUI/qt_resources/mainwindow.ui]: Why are the width and height of this (seemingly) unrelated page changed?

File: TriblerGUI/tribler_window.py [hunk 2]
```diff
@@ -134,7 +134,14 @@ def __init__(self):
         self.loading_page.initialize_loading_page()
         self.discovering_page.initialize_discovering_page()
         self.discovered_page.initialize_discovered_page()
-        self.trust_page.initialize_trust_page()
+
+        # If QWebEngineView cannot be imported, make sure to render the old trust page instead.
```
Comment (vandenheuvel) [inline TriblerGUI/tribler_window.py]: Please rename "old trust page" to "graph trust page".

File: TriblerGUI/tribler_app.py [hunk 3]
```diff
@@ -3,7 +3,11 @@
 from PyQt5.QtCore import QEvent
 from TriblerGUI.single_application import QtSingleApplication
 # QtWebEngineWidgets must be imported before a QCoreApplication instance is created, otherwise an import error occurs
-from PyQt5.QtWebEngineWidgets import QWebEngineView
+try:
+    from PyQt5.QtWebEngineWidgets import QWebEngineView
+except ImportError:
+    # In the case QWebEngineView is not available, render the old trust page.
```
Comment (vandenheuvel) [inline TriblerGUI/tribler_app.py]: Please rename "old trust page" to "graph trust page".

File: TriblerGUI/tribler_window.py [hunk 4]
```diff
@@ -134,7 +134,14 @@ def __init__(self):
...
+        try:
+            from PyQt5.QtWebEngineWidgets import QWebEngineView
+            self.trust_page_network.initialize_trust_page()
+        except ImportError:
+            self.trust_page_graph.initialize_trust_page()
+            self.left_menu_button_trust_display.clicked.connect(self.clicked_menu_button_trust_old)
```
Comment (vandenheuvel) [inline TriblerGUI/tribler_window.py]: Please rename "old trust page" to "graph trust page".

File: TriblerGUI/tribler_window.py [hunk 5]
```diff
@@ -550,7 +551,14 @@ def clicked_menu_button_subscriptions(self):
 
     def clicked_menu_button_trust(self):
         self.deselect_all_menu_buttons(self.left_menu_button_trust_display)
-        self.stackedWidget.setCurrentIndex(PAGE_TRUST)
+        self.stackedWidget.setCurrentIndex(PAGE_TRUST_NETWORK)
+        self.navigation_stack = []
+        self.hide_left_menu_playlist()
+
+    def clicked_menu_button_trust_old(self):
```
Comment (vandenheuvel) [inline TriblerGUI/tribler_window.py]: Please rename "old trust page" to "graph trust page".

File: TriblerGUI/widgets/trustpagenetwork.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+            # The path to the main html file
+            path = os.path.join(utilities.get_base_path(), "widgets/trustpage/index.html")
+
+            view.setUrl(QUrl.fromLocalFile(path))
+            view.page().setBackgroundColor(QColor.fromRgb(0, 0, 0, 0))
+            view.show()
+
+            vertical_layout.addWidget(view)
+        except ImportError:
+            # In the case QWebEngineView is not available, render the old trust page.
```
Comment (vandenheuvel) [inline TriblerGUI/widgets/trustpagenetwork.py]: Please rename "old trust page" to "graph trust page".

**Comment by MaxVanDeursen** [comment]
@vandenheuvel Please have another look at this PR, especially the last commit

File: TriblerGUI/tribler_window.py [hunk 7]
```diff
@@ -142,6 +142,12 @@ def __init__(self):
             self.trust_page_network.initialize_trust_page()
         except ImportError:
             self.trust_page_graph.initialize_trust_page()
+            label = self.stackedWidget.widget(PAGE_TRUST_GRAPH).window().trust_explanation_label
+            label_text = label.text()
+            text_split = label_text.split("LINK_PLACEHOLDER")
+            index_html = os.path.dirname(os.path.abspath(__file__)) + "/widgets/trustpage/index.html"
```
Comment (vandenheuvel) [inline TriblerGUI/tribler_window.py]: Can you use `os.path.join`?

File: TriblerGUI/widgets/trustpagenetwork.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        Load the pyplot graph into the QWidget.
+        """
+        try:
+            from PyQt5.QtWebEngineWidgets import QWebEngineView
+            vertical_layout = self.window().network_widget.layout()
+
+            view = QWebEngineView()
+
+            # The path to the main html file
+            path = os.path.join(utilities.get_base_path(), "widgets/trustpage/index.html")
```
Comment (vandenheuvel) [inline TriblerGUI/widgets/trustpagenetwork.py]: Please create a single method to find the `index.html` file.

**Comment by MaxVanDeursen** [comment]
retest this please

**Comment by vandenheuvel** [comment]
retest this please

**Comment by vandenheuvel** [comment]
retest this please

**Pull request merged**